src/modules/show/commentPopup.js

```diff
@@ -0,0 +1,151 @@
+import Store from '../store.js';
+import CommentApi from '../comments/commentApi.js';
+import Comment from '../comment/comment.js';
+
+export default class CommentPopup {
+  constructor(selectedItem) {
+    this.selectedItem = selectedItem;
+    this.store = new Store();
+    this.commentApi = new CommentApi();
+    this.nameInput = null;
+    this.commentTextArea = null;
+  }
+
+  async loadPopup() {
+    const showDetails = await this.fetchShowDetails();
+
+    if (!showDetails || !showDetails.image || !showDetails.name) {
+      throw Error('Failed to fetch show details or show image or name is undefined');
+    }
+    const {
+      name, image, language, genres, schedule,
+    } = showDetails;
+    if (!name || !image) {
+      throw Error('Show name or image is undefined');
+    }
+    const popupContainer = document.createElement('div');
+    popupContainer.classList.add('popup-container');
+
+    popupContainer.style.backgroundColor = '#333';
+    popupContainer.style.position = 'fixed';
+    popupContainer.style.top = '50%';
+    popupContainer.style.left = '50%';
+    popupContainer.style.border = '1px solid #f1f1f1';
+    popupContainer.style.transform = 'translate(-50%, -50%)';
+    popupContainer.style.zIndex = '9999';
+
+    const closeButton = document.createElement('button');
+    closeButton.classList.add('close-button');
+    closeButton.textContent = 'X';
+    popupContainer.appendChild(closeButton);
+
+    const showImage = document.createElement('img');
+    showImage.src = image.medium;
+    popupContainer.appendChild(showImage);
+
+    const showName = document.createElement('h1');
+    showName.textContent = name;
+    popupContainer.appendChild(showName);
+
+    const sContent = document.createElement('div');
+    sContent.classList.add('s-content');
+    const showlanguage = document.createElement('h4');
+    showlanguage.textContent = `Language: ${language}`;
+    sContent.appendChild(showlanguage);
+
+    const showgenres = document.createElement('h4');
+    showgenres.textContent = `Geners: ${genres}`;
+    sContent.appendChild(showgenres);
+
+    const showSchedule = document.createElement('h4');
+    showSchedule.textContent = `Schedule: ${schedule.days} at ${schedule.time}`;
+    sContent.appendChild(showSchedule);
+    popupContainer.appendChild(sContent);
+    const commentsContainer = document.createElement('div');
+    commentsContainer.classList.add('comments-container');
+    popupContainer.appendChild(commentsContainer);
+    const title = document.createElement('h3');
+    title.textContent = 'Add a Comment';
+    popupContainer.appendChild(title);
+    this.nameInput = document.createElement('input');
```

Comment:
- [ ] It would be better if we style our comment popup text properly, some of the text is missing as we can see in the screenshot below
![Screenshot from 2023-05-25 20-19-37](https://github.com/jojo987N/Moviehub-Webapp/assets/46388113/b67cf70c-44ee-469a-a1f0-1ee46e8e45f1)